### PR TITLE
Run AbilityCast triggers for mana abilities

### PR DESCRIPTION
**Pull request by tool4ever**
Test case:
Enchant _Crystalline Crawler_ with _Artifact Possession_

When you generate mana by removing a counter it should trigger the Aura

It also matters for the new _Vazi, Keen Negotiator_.

- [x] need to update cards like _Harsh Mentor_ with ValidSA to filter out mana abilities

### CODE REVIEW

**Comment by Hanmac** [comment]
Need to check if such trigger would stop the mana ability from being undoable

**Comment by tool4ever** [comment]
Good point.

Added a block to catch it.

Also fixed related bugs:
- if you activated _Sol Ring_ with _Forsaken Monument_ in play, if it LTB later the mana ability would remain undoable

- the way ManaReflected with _Winter's Night_ worked made it Pump it twice

File: forge-game/src/main/java/forge/game/ability/effects/ManaReflectedEffect.java [hunk 1]
```diff
@@ -23,16 +23,14 @@ public class ManaReflectedEffect extends SpellAbilityEffect {
     public void resolve(SpellAbility sa) {
         // Spells are not undoable
         AbilityManaPart ma = sa.getManaPart();
-        sa.setUndoable(sa.isAbility() && sa.isUndoable());
+        sa.setUndoable(sa.isAbility() && sa.isUndoable() && sa.getSubAbility() == null);
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/ability/effects/ManaReflectedEffect.java:26]: what if it had two+ mana abilities as chain?
Like add Mana, and then add Additional Mana?

or doesn't this matter for Mana Reflected?

File: forge-game/src/main/java/forge/game/trigger/TriggerSpellAbilityCastOrCopy.java [hunk 2]
```diff
@@ -90,14 +90,23 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
...
+            Player activator;
+            if (spellAbility.isManaAbility()) {
+                activator = (Player) runParams.get(AbilityKey.Activator);
+            } else if (si == null) {
+                return false;
+            } else {
+                activator = si.getSpellAbility(true).getActivatingPlayer();
+            }
+
+            if (!matchesValid(activator, getParam("ValidActivatingPlayer").split(","))) {
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/trigger/TriggerSpellAbilityCastOrCopy.java:102]: `matchesValidParam("ValidActivatingPlayer", activator)`

File: forge-game/src/main/java/forge/game/ability/effects/ManaReflectedEffect.java (hunk 1 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/effects/ManaReflectedEffect.java:26]: Couldn't find any like that
also ManaEffect will do the same for subs (in `GameActionUtil.generatedMana`)

**Review by Hanmac — APPROVED** [review]
:+1:

**Pull request merged**